### PR TITLE
Fix #39462

### LINKED ISSUE #39462: ReduceLROnPlateau keeps executing lr reduction block of code after min_lr has been reached
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: None
- TensorFlow installed from (source or binary): through pip
- TensorFlow version (use command below): 2.2
- Python version: 3.7
- Bazel version (if compiling from source): n/a
- GCC/Compiler version (if compiling from source): n/a
- CUDA/cuDNN version: n/a
- GPU model and memory: n/a

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

ReduceLR will execute the part of the code that reduces LR even if lr "equals" min_lr. While this is not a problem if we are just dealing with LR since the value will technically never go below min_lr, it is an issue if you are trying to do any other execution in this block of code (ie. weight decay).

Given that there is a min check for this block of code, i assume it was not intended for us to enter this block of code if min_lr has already been achieved. Either get rid of the min check if we don't care about re-executing this code for no reason, or make sure this block of code is not executed if lr is at min-lr.

This issue clearly seems to be a round/precision related issue where 
we set our lr `K.set_value(self.model.optimizer.lr, new_lr)`

But then next iteration we fetch the lr
`old_lr = float(K.get_value(self.model.optimizer.lr))`

and get a slightly different value so that 
`if old_lr > self.min_lr:`

does not work.

So yes in a way this does not affect default code, and it only matter if we do custom code, but i still think this is a bug that really shouldn't happen.

**Describe the expected behavior**

Code block that is only supposed to execute if lr does not equal min_lr does not execute if lr equals min_lr

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

```
x = np.random.normal(size=10)
y = np.random.normal(size=10)

model = Sequential()
model.add(Dense(1))

reduce_lr = ReduceLROnPlateau(monitor='loss',
                              min_delta=0.01,
                              patience=3,
                              min_lr=0.001,
                              verbose=2)

model.compile(loss='mse', optimizer=SGD(0.01))

history = model.fit(x,
                    y,
                    epochs=25,
                    verbose=2,
                    shuffle=True,
                    batch_size=1,
                    callbacks=[reduce_lr])
```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

### PR DESCRIPTION
**Pull request by ben-arnao**
When using `get_value` to retrieve the value for LR, we receive a np.float32. However when we cast this to a python float, there are unwanted trailing significant digits added to our value that causes old_lr to never be exactly equal to our supplied min_lr, despite it being equal internally.

By leaving old_lr as it's original np.float32, and by casting our supplied min_lr to np.float32, we are now able to properly compare the values and the block of code should not be executed again after lr has been set to min_lr.

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
LGTM, thank you for the PR, and sorry for the very long delay in reviewing.
